File: Algorithms/src/2147. Number of Ways to Divide a Long Corridor/Code/solution.cpp
class Solution
{
public:
    int numberOfWays(string corridor)
    {
        const int MOD = 1e9 + 7;
        vector<int> seats;

        // Store indices of all seats
        for (int i = 0; i < corridor.size(); i++)
        {
            if (corridor[i] == 'S')
            {
                seats.push_back(i);
            }
        }

        int total = seats.size();
        if (total == 0 || total % 2 != 0)
            return 0;

        long long ways = 1;

        // Multiply gaps between sections
        for (int i = 2; i < total; i += 2)
        {
            ways = (ways * (seats[i] - seats[i - 1])) % MOD;
        }

        return ways;
    }
};

File: Algorithms/src/2147. Number of Ways to Divide a Long Corridor/Code/solution.js
var numberOfWays = function (corridor) {
  const MOD = 1_000_000_007;
  const seats = [];

  for (let i = 0; i < corridor.length; i++) {
    if (corridor[i] === "S") seats.push(i);
  }

  if (seats.length === 0 || seats.length % 2 !== 0) return 0;

  let ways = 1;
  for (let i = 2; i < seats.length; i += 2) {
    ways = (ways * (seats[i] - seats[i - 1])) % MOD;
  }

  return ways;
};
